src/client.cpp

```diff
@@ -956,6 +999,73 @@ void Client::sendLegacyInit(const char* playerName, const char* playerPassword)
 	Send(&pkt);
 }
 
+void Client::sendInit(const std::string &playerName)
+{
+	NetworkPacket pkt(TOSERVER_INIT, 1 + 2 + 2 + (1 + playerName.size()));
+
+	// TODO (later) actually send supported compression modes
+	pkt << (u8) SER_FMT_VER_HIGHEST_READ << (u8) 42;
+	pkt << (u16) CLIENT_PROTOCOL_VERSION_MIN << (u16) CLIENT_PROTOCOL_VERSION_MAX;
+	pkt << playerName;
+
+	Send(&pkt);
+}
+
+void Client::startAuth(AuthMechanism chosen_auth_mechanism)
+{
+	m_chosen_auth_mech = chosen_auth_mechanism;
+
+	switch (chosen_auth_mechanism) {
+		case AUTH_MECHANISM_FIRST_SRP: {
+			m_authstate = 1;
+			// send srp verifier to server
+			NetworkPacket resp_pkt(TOSERVER_FIRST_SRP, 0);
+			const char *salt, *bytes_v;
+			int len_salt, len_v;
+			salt = NULL;
+			getSRPVerifier(getName(), m_password, &salt,
+				&len_salt, &bytes_v, &len_v);
+			resp_pkt.putCString(salt, len_salt);
+			resp_pkt.putCString(bytes_v, len_v);
+			free((char *) salt);
+			free((char *) bytes_v);
+			resp_pkt << ((m_password == "") ? (u8) 1 : (u8) 0);
```

Comment:
Casting the whole expression might be neater. e.g. `resp_pkt << (u8)(m_password == "" ? 1 : 0);